### ISSUE TITLE
Error starting daemon: couldn't create plugin manager: error setting plugin manager root to private: permission denied

### ISSUE THREAD

**Issue by BirkhoffLee** [moby/moby#35315]
**Description**

**Steps to reproduce the issue:**
1. Have a brand new server set up (CentOS 7 minimal)
2. Run the commands below w/ root/sudo permission

```
$ yum install -y yum-utils device-mapper-persistent-data lvm2
$ yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
$ yum install docker-ce -y
$ service docker start
```

**Describe the results you received:**
```
[root@Birkhoff-VPS ~]# service docker restart
Redirecting to /bin/systemctl restart docker.service
Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.

[root@Birkhoff-VPS ~]# journalctl -xe
...
Oct 27 08:39:20 Birkhoff-VPS dockerd[29243]: time="2017-10-27T08:39:20.206015329Z" level=info msg="libcontainerd: new containerd process, pid: 29249"
Oct 27 08:39:21 Birkhoff-VPS dockerd[29243]: Error starting daemon: couldn't create plugin manager: error setting plugin manager root to private: permission denied
...
```

**Describe the results you expected:**
The dockerd launches successfully

**Additional information you deem important (e.g. issue happens only occasionally):**

There are some other people also encountered the exactly same problem **recently**, so I guess this is a bug in a recent update.

Links to the related discussions/questions:
1. https://forums.docker.com/t/cannot-install-docker-on-fresh-ubuntu-17-server-docker-is-missing-permissions-to-operate-in-var-lib-docker/39852
2. https://stackoverflow.com/questions/46961918/error-starting-docker-daemon-on-centos-7

**Output of `docker version`:**

```
[root@Birkhoff-VPS ~]# docker version
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:53:49 2017
 OS/Arch:      linux/amd64
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Output of `docker info`:**

```
[root@Birkhoff-VPS ~]# docker info
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

```
[root@Birkhoff-VPS ~]# uname -a
Linux Birkhoff-VPS 4.10.17-2-pve #1 SMP PVE 4.10.17-19 (Fri, 4 Aug 2017 13:34:37 +0200) x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by raphaklaus** [comment]
Are you using CentOS on top of LXC?

**Comment by BirkhoffLee** [comment]
@raphaklaus Very likely. Does that have any impact on this?

**Comment by raphaklaus** [comment]
Yes. Maybe is related to security nesting setup. I manage to get it working using LXD instead setting `security.nesting` to true

**Comment by tnguyenns1** [comment]
Met this issue, but it seems to be fixed via version 18.03.0:

Ref: https://docs.docker.com/release-notes/docker-ce/#18030-ce-2018-03-21

> Perform plugin mounts in the runtime moby/moby#35829

**Comment by cpuguy83** [comment]
Yes, the code that caused this does not exist anymore.
Closing.

Thanks!
